test/BlockChainHelper.cpp

```diff
@@ -0,0 +1,396 @@
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file BlockChainHelper.cpp
+ * @author Dimitry Khokhlov <dimitry@ethdev.com>
+ * @date 2015
+ */
+
+#include <libethereum/Block.h>
+#include <libethereum/BlockChain.h>
+#include <libethereum/TransactionQueue.h>
+#include <libdevcore/TransientDirectory.h>
+#include <test/BlockChainHelper.h>
+#include <test/TestHelper.h>
+
+using namespace std;
+using namespace json_spirit;
+using namespace dev;
+using namespace dev::eth;
+
+namespace dev {  namespace test {
+
+TestTransaction::TestTransaction(mObject const& _o): m_jsonTransaction(_o)
+{
+	ImportTest::importTransaction(_o, m_transaction); //check that json structure is valid
+}
+
+TestBlock::TestBlock(TestBlock const& _original)
+{
+	populateFrom(_original);
+}
+
+TestBlock& TestBlock::operator = (TestBlock const& _original)
+{
+	populateFrom(_original);
+	return *this;
+}
+
+TestBlock::TestBlock(mObject const& _blockObj, mObject const& _stateObj, RecalcBlockHeader _verify)
+{
+	m_state = State(OverlayDB(State::openDB(m_tempDirState.path(), h256{}, WithExisting::Kill)), BaseState::Empty);
+	ImportTest::importState(_stateObj, m_state);
+	m_state.commit();
+
+	m_blockHeader = constructBlock(_blockObj, _stateObj.size() ? m_state.rootHash() : h256{});
+	recalcBlockHeaderBytes(_verify);
+}
+
+TestBlock::TestBlock(std::string const& _blockRlp)
+{
+	m_bytes = importByteArray(_blockRlp);
+
+	RLP root(m_bytes);
+	m_blockHeader.populateFromHeader(root[0], IgnoreSeal);
+
+	m_transactionQueue.clear();
+	m_testTransactions.clear();
+	for (auto const& tr: root[1])
+	{
+		Transaction tx(tr.data(), CheckTransaction::Everything);
+		TestTransaction testTx(tx);
+		m_transactionQueue.import(tx.rlp());
+		m_testTransactions.push_back(testTx);
+	}
+
+	for	(auto const& uRLP: root[2])
+	{
+		BlockHeader uBl;
+		uBl.populateFromHeader(uRLP);
+		TestBlock uncle;
+		//uncle goes without transactions and uncles but
+		//it's hash could contain hashsum of transactions/uncles
+		//thus it won't need verification
+		uncle.setBlockHeader(uBl, RecalcBlockHeader::SkipVerify);
+		m_uncles.push_back(uncle);
+	}
+
+	//copyStateFrom(_original.getState());
+}
+
+void TestBlock::setState(State const& _state)
+{
+	copyStateFrom(_state);
+}
+
+void TestBlock::addTransaction(TestTransaction const& _tr)
+{
+	try
+	{
+		m_testTransactions.push_back(_tr);
+		if (m_transactionQueue.import(_tr.getTransaction().rlp()) != ImportResult::Success)
+			cnote << "Test block failed importing transaction\n";
+	}
+	catch (Exception const& _e)
+	{
+		BOOST_ERROR("Failed transaction constructor with Exception: " << diagnostic_information(_e));
+	}
+	catch (exception const& _e)
+	{
+		cnote << _e.what();
+	}
+}
+
+void TestBlock::addUncle(TestBlock const& _uncle)
+{
+	m_uncles.push_back(_uncle);
```

Comment:
there a couple of functions only containing one line of code,why not use `m_uncles.push_back(_uncle);` directly instead of `addUncle(_uncle);` ?